### PR TITLE
Import badges csv

### PR DESCRIPTION
**Pull request by ZePedroResende**
Format:

Coffee break;badge do lanche;2019-02-11;2019-02-12;/tmp/goraster.png

### CODE REVIEW

File: lib/mix/tasks/gen.badges.ex [hunk 1]
```diff
@@ -0,0 +1,51 @@
+defmodule Mix.Tasks.Gen.Badges do
+  use Mix.Task
+  import Mix.Ecto
+
+  NimbleCSV.define(MyParser, separator: ";", escape: "\"")
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Most editors use ``,``instead of ``;`` any reason for this change?

File: lib/mix/tasks/gen.badges.ex [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  # format Coffee break;badge do lanche;2019-02-11;2019-02-12;/tmp/goraster.png
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Does the code accept a header in the ``CSV``? If so also add that here

File: lib/mix/tasks/gen.badges.ex [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  def run(args) do
+    cond do
+      length(args) == 0 ->
+        Mix.shell.info "Needs to receive a number greater than 0."
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Is this info correct?

File: lib/mix/tasks/gen.badges.ex [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      end)
+  end
+
+  defp parse_csv(path) do
+    path
+    |> File.stream!
+    |> MyParser.parse_stream
+    |> Stream.map(
+      fn [name, description,begin_time,end_time,url] ->
+        {:ok,b_t,_} = DateTime.from_iso8601("#{begin_time}T00:00:00Z")
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Add spaces and use more useful names

File: lib/mix/tasks/gen.badges.ex [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        {:ok,e_t,_} = DateTime.from_iso8601("#{end_time}T00:00:00Z")
+
+        %{name: name,
+          description: description,
+          begin: b_t,
+          end: e_t,
+          avatar: url}
+      end)
+  end
+
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Remove extra lines

File: lib/mix/tasks/gen.badges.ex (hunk 1 above)
Comment (ZePedroResende) [reply lib/mix/tasks/gen.badges.ex]: excel generated this with ; by default , we can pass de separator as a argument if you think its trully necessary

File: lib/mix/tasks/gen.badges.ex (hunk 3 above)
Comment (0urobor0s) [reply lib/mix/tasks/gen.badges.ex]: Do you receive the csv as argument so wouldn't the message be other thing?

File: lib/mix/tasks/gen.badges.ex [hunk 6]
```diff
@@ -9,7 +9,7 @@ defmodule Mix.Tasks.Gen.Badges do
   def run(args) do
     cond do
       length(args) == 0 ->
-        Mix.shell.info "Needs to receive a number greater than 0."
+        Mix.shell.info "Needs to receive atleast one file path ."
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Missing space at ``at least``

File: lib/mix/tasks/gen.badges.ex [hunk 7]
```diff
@@ -36,16 +36,14 @@ defmodule Mix.Tasks.Gen.Badges do
     |> MyParser.parse_stream
     |> Stream.map(
       fn [name, description,begin_time,end_time,url] ->
-        {:ok,b_t,_} = DateTime.from_iso8601("#{begin_time}T00:00:00Z")
-        {:ok,e_t,_} = DateTime.from_iso8601("#{end_time}T00:00:00Z")
+        {:ok,begin_datetime,_} = DateTime.from_iso8601("#{begin_time}T00:00:00Z")
```
Comment (0urobor0s) [inline lib/mix/tasks/gen.badges.ex]: Missing space after commas

**Pull request merged**